pkg/sql/alisa_submitter.go

```diff
@@ -61,74 +60,58 @@ func (s *alisaSubmitter) submitAlisaTask(code, resourceName string) error {
 	return e
 }
 
-func (s *alisaSubmitter) getModelPath(modelName string) (string, error) {
-	_, dsName, err := database.ParseURL(s.Session.DbConnStr)
-	if err != nil {
-		return "", err
-	}
-	cfg, err := goalisa.ParseDSN(dsName)
-	if err != nil {
-		return "", err
-	}
-	userID := s.Session.UserId
-	if userID == "" {
-		userID = "unkown"
-	}
-	return strings.Join([]string{cfg.Project, userID, modelName}, "/"), nil
-}
-
 func (s *alisaSubmitter) ExecuteTrain(ts *ir.TrainStmt) (e error) {
 	ts.TmpTrainTable, ts.TmpValidateTable, e = createTempTrainAndValTable(ts.Select, ts.ValidationSelect, s.Session.DbConnStr)
 	if e != nil {
 		return e
 	}
 	defer dropTmpTables([]string{ts.TmpTrainTable, ts.TmpValidateTable}, s.Session.DbConnStr)
 
-	cc, e := pai.GetClusterConfig(ts.Attributes)
+	ossModelPath, e := getModelPath(ts.Into, s.Session)
 	if e != nil {
 		return e
 	}
 
-	modelPath, e := s.getModelPath(ts.Into)
-	if e != nil {
+	if e = s.cleanUpModel(ossModelPath); e != nil {
 		return e
 	}
-
-	paiCmd, e := getPAIcmd(cc, ts.Into, modelPath, ts.TmpTrainTable, ts.TmpValidateTable, "")
+	// upload Alisa resource file to OSS
+	ossObjectName := randStringRunes(16)
+	// Alisa resource should be prefix with @@, alisa source would replace it with the RES_DOWN_URL.resourceName in alisa env.
+	scriptPath := fmt.Sprintf("file://@@%s", resourceName)
+	code, paiCmd, e := pai.Train(ts, s.Session, scriptPath, ts.Into, ossModelPath, s.Cwd)
 	if e != nil {
 		return e
 	}
+	return s.alisaCmd(code, paiCmd, ossObjectName)
+}
 
-	code, e := pai.TFTrainAndSave(ts, s.Session, modelPath, cc)
-	if e != nil {
-		return e
-	}
+func (s *alisaSubmitter) alisaCmd(program, alisaCode, ossObjectName string) error {
```

Comment:
`alisaCmd` does uploading code resource and run the submit. How about splitting these two operations to two functions?

Reply:
I renamed it to `uploadResourceAndSubmitAlisaTask`, maybe it's a common function can be used in `ExecuteTrain` , `ExecutePredict` and `ExecuteExplain`